In [1]:
import ollama

# Ollamaサーバーから利用可能なモデルのリストを取得します
# response_objectにはモデル情報を含むオブジェクトが返されます
response_object = ollama.list()

# 'models'属性があり、かつモデルリストが空でないかを確認します
if hasattr(response_object, 'models') and response_object.models:
    # 利用可能なモデル名のみを出力します
    for model_data in response_object.models:
        # model_dataオブジェクトからモデル名（.model属性）を取得して表示します
        print(model_data.model)
else:
    # モデルが見つからなかった場合のメッセージ
    print("利用可能なモデルが見つかりませんでした。Ollamaサーバーが起動しており、モデルがインストールされているか確認してください。")

deepseek-r1:8b
llama3.1:8b
qwen2.5-coder:7b


In [ ]:
import ollama

# --- 設定項目 ---
# 使用するモデル名を指定します (1セル目で確認したモデル名を入力してください)
MODEL_TO_USE = 'deepseek-r1:8b'  # 例: 'gemma3:1b', 'llama3'など

# LLMへの指示（プロンプト）
USER_PROMPT = "あなたのできることを教えて下さい"

# (オプション) システムプロンプトでAIの基本的な振る舞いを指示できます
# ここでは非常にシンプルな指示にしています
SYSTEM_PROMPT = "あなたはAIアシスタントです。簡潔に答えてください。"
# --- 設定ここまで ---

# LLMにチャット形式で指示を送信します
response = ollama.chat(
    model=MODEL_TO_USE,
    messages=[
        {
            'role': 'system',
            'content': SYSTEM_PROMPT
        },
        {
            'role': 'user',
            'content': USER_PROMPT
        }
    ]
)

# LLMからの応答メッセージの内容のみを出力します
print(response['message']['content'])

In [1]:
# deepseek-r1:8b, llama3.1:8b, qwen2.5-coder:7b のそれぞれの応答速度を確かめる
import time
import ollama

# --- 設定 ---
MODELS = [
    'deepseek-r1:8b',
    'llama3.1:8b',
    'qwen2.5-coder:7b',
]
SYSTEM_PROMPT = "あなたはAIアシスタントです。簡潔に答えてください。"
USER_PROMPT = "あなたのできることを教えて下さい"
RUNS_PER_MODEL = 3  # 1モデルあたり何回計測するか
# --- 設定ここまで ---

def measure_model(model_name):
    times = []
    answers = []
    for i in range(RUNS_PER_MODEL):
        start = time.time()
        response = ollama.chat(
            model=model_name,
            messages=[
                {'role': 'system', 'content': SYSTEM_PROMPT},
                {'role': 'user',   'content': USER_PROMPT},
            ]
        )
        elapsed = time.time() - start
        times.append(elapsed)
        answers.append(response['message']['content'])
        print(f"[{model_name} Run {i+1}] {elapsed:.3f} 秒")
    avg_time = sum(times) / len(times)
    print(f"→ {model_name} の平均応答時間: {avg_time:.3f} 秒\n")
    return {
        'model': model_name,
        'times': times,
        'avg_time': avg_time,
        'answers': answers,
    }

def main():
    results = []
    print("=== 応答速度 & 回答内容 比較 ===\n")
    for m in MODELS:
        print(f"--- モデル: {m} ---")
        try:
            res = measure_model(m)
            results.append(res)
        except Exception as e:
            print(f"Error for {m}: {e}\n")
    # 結果をまとめて表示
    print("=== まとめ ===")
    for r in results:
        print(f"\nモデル: {r['model']}")
        print(f"平均時間: {r['avg_time']:.3f} 秒")
        print("【回答例】")
        for idx, ans in enumerate(r['answers'], 1):
            print(f"  Run {idx}: {ans[:100]}{'…' if len(ans)>100 else ''}")
    # 必要に応じてここで自動評価指標を計算してください

if __name__ == '__main__':
    main()



=== 応答速度 & 回答内容 比較 ===

--- モデル: deepseek-r1:8b ---
[deepseek-r1:8b Run 1] 58.117 秒
[deepseek-r1:8b Run 2] 140.283 秒
[deepseek-r1:8b Run 3] 126.096 秒
→ deepseek-r1:8b の平均応答時間: 108.165 秒

--- モデル: llama3.1:8b ---
[llama3.1:8b Run 1] 24.420 秒
[llama3.1:8b Run 2] 15.763 秒
[llama3.1:8b Run 3] 17.661 秒
→ llama3.1:8b の平均応答時間: 19.281 秒

--- モデル: qwen2.5-coder:7b ---
[qwen2.5-coder:7b Run 1] 37.433 秒
[qwen2.5-coder:7b Run 2] 26.084 秒
[qwen2.5-coder:7b Run 3] 21.571 秒
→ qwen2.5-coder:7b の平均応答時間: 28.363 秒

=== まとめ ===

モデル: deepseek-r1:8b
平均時間: 108.165 秒
【回答例】
  Run 1: <think>
よし、ユーザーから「あなたのできることを教えて下さい」という質問が来た。これは私の機能を簡単に説明してほしいという意味だ。

まず、私が日本語で対応できるAIアシスタントであることを伝…
  Run 2: <think>
ふむ、ユーザーが「あなたのできることを教えて下さい」と質問してきたんだな。これは初めてのやり取りか、あるいは機能確認したいのか。どちらにせよ、明確で有用な説明が必要だ。

AIアシスタ…
  Run 3: <think>
ふむ、ユーザーが「あなたのできることを教えて下さい」と尋ねてきた。これは初めてのやり取りか、あるいは機能確認としての一般的な質問だろう。AIアシスタントとしての基本的な役割説明を求めて…

モデル: llama3.1:8b
平均時間: 19.281 秒
【回答例】
  Run 1: 以下が可能です。

*   質問に答える
*   情報を提供する
*   計算を行う
*   テキストの作成と修正を行う
*

### 応答速度比較結果

| モデル                | Run 1 (秒) | Run 2 (秒) | Run 3 (秒) | 平均応答時間 (秒) |
|:---------------------|-----------:|-----------:|-----------:|-----------------:|
| deepseek-r1:8b       |     58.117 |    140.283 |    126.096 |           108.165 |
| llama3.1:8b          |     24.420 |     15.763 |     17.661 |            19.281 |
| qwen2.5-coder:7b     |     37.433 |     26.084 |     21.571 |            28.363 |
